siliconcompiler/core.py

```diff
@@ -3552,6 +3569,19 @@ def pipe(self, flow, plan):
                 self.edge(flow, prevstep, step)
             prevstep = step
 
+    ###########################################################################
+    def __write_task_manifest(self, tool, path=None, backup=True):
```

Comment:
Is a two underscore prefix a distinct meaning from one underscore?

Reply:
__ gets renamed at runtime to __chip_somethign. This ensures private variables are very difficult to access, while _ is just by convention private, __ is a little more private because the name gets modified by the file it's coming from. This prevents users from modifying the relative path this even by accident. https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-single-and-double-underscore-before-an-object-name

Reply:
Ah yeah, this is vaguely familiar. Thanks for the explanation! I imagine it would make sense to move over to double underscore in a lot of other cases/adopt it for things going forward as you've done here.